src/oc/storage/api/boards.clj

```diff
@@ -162,46 +164,54 @@
-            (timbre/info "Upserted entry for new board:" board-uuid "as" (:uuid entry-result))
-            (when (= (:status entry-result) "published")
-              (when (= :add entry-action)
-                (entries-api/auto-share-on-publish conn (assoc ctx :existing-board board-result) entry-result))
-              (notification/send-trigger! (notification/->trigger entry-action org board-result {:new entry-result} (:user ctx) nil))))))
-      (let [created-board (if (and (empty? authors) (empty? viewers))
-                            ;; no additional members added, so using the create response is good
-                            board-result
-                            ;; retrieve the board again to get final list of members
-                            (board-res/get-board conn (:uuid board-result)))]
-        (notification/send-trigger! (notification/->trigger :add org {:new created-board :notifications notifications} user invitation-note))
-        {:created-board created-board}))
+  (let [entries (:entries new-board)
+        draft-board? (and (pos? (count entries)) (every? #(-> % :status keyword (= :draft)) entries))
+        new-board-data (assoc new-board :draft draft-board?)]
+    (timbre/info "Crating board, is draft?" draft-board?)
+    (if-let [board-result (board-res/create-board! conn new-board-data)] ; Add the board
+
+      (let [board-uuid (:uuid board-result)
+            authors (-> ctx :data :authors)
+            viewers (-> ctx :data :viewers)
+            invitation-note (-> ctx :data :note)
+            notifications (:notifications ctx)]
+        (timbre/info "Created board:" board-uuid "for org:" org-slug)
+        ;; Add any authors specified in the request
+        (doseq [author authors] (add-member conn ctx (:slug org) (:slug board-result) :authors author))
+        ;; Add any viewers specified in the request
+        (doseq [viewer viewers] (add-member conn ctx (:slug org) (:slug board-result) :viewers viewer))
+        ;; Add any entries specified in the request
+        (doseq [entry entries]
+          (let [fixed-entry (-> entry
+                                (assoc :status (or (:status entry) "published"))
+                                (assoc :board-uuid board-uuid))
+                entry-action (if (entry-res/get-entry conn (:uuid entry))
+                               :update
+                               :add)
+                new-entry (if (= entry-action :update)
+                            fixed-entry
+                            (entry-res/->entry conn board-uuid fixed-entry user))]
+            (timbre/info "Upserting entry for new board:" board-uuid)
+            (let [entry-result (entry-res/upsert-entry! conn new-entry user)]
+              (timbre/info "Upserted entry for new board:" board-uuid "as" (:uuid entry-result))
+              ;; If we are updating an existing draft check if we need to remove the old board
```

Comment:
This code should also be moved into the board resources module instead of here in API.

Reply:
This code was already here, i only indented it.. i think you are suggesting a bit a "refactor" here. Move all the logic behind the api call to the resource namespace where the resources here are only DB connectors without any intelligence or logic.